holmes/plugins/toolsets/database/database.py

```
@@ -0,0 +1,573 @@
+    """Rewrite a connection URL to use a pure-Python driver when possible."""
+    parsed = urlparse(raw_url)
+    scheme = parsed.scheme  # e.g. "postgresql", "mysql+pymysql", "postgres"
+
+    for prefix, replacement in _DRIVER_MAP.items():
+        if scheme == prefix or scheme.startswith(prefix + "+"):
+            # Only replace if the user hasn't already picked the right driver
+            if scheme != replacement:
+                return raw_url.replace(scheme, replacement, 1)
+            return raw_url
+
+    # Unknown scheme – return as-is and let SQLAlchemy handle it
```

Ruff (0.15.1)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)